src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -28,25 +28,43 @@ public TypeFactory(OutputLibrary library)
             _library = library;
         }
 
-        public CSharpType CreateType(Schema schema, bool isNullable) => CreateType(schema, schema.Extensions, isNullable);
+        public CSharpType CreateType(InputType inputType) => inputType.Kind switch
+        {
+            InputTypeKind.AzureLocation => new CSharpType(typeof(AzureLocation), inputType.IsNullable),
+            InputTypeKind.Boolean => new CSharpType(typeof(bool), inputType.IsNullable),
+            InputTypeKind.Bytes => new CSharpType(typeof(byte[]), inputType.IsNullable),
+            InputTypeKind.DateTime => new CSharpType(typeof(DateTimeOffset), inputType.IsNullable),
+            InputTypeKind.Dictionary => new CSharpType(typeof(IDictionary<,>), inputType.IsNullable, typeof(string), CreateType(inputType.ValuesType!)),
+            InputTypeKind.Enum => CreateType(inputType.ValuesType! with { IsNullable = inputType.IsNullable }),
+            InputTypeKind.ETag => new CSharpType(typeof(ETag), inputType.IsNullable),
+            InputTypeKind.ExtensibleEnum => CreateType(inputType.ValuesType! with { IsNullable = inputType.IsNullable }),
+            InputTypeKind.Float32 => new CSharpType(typeof(float), inputType.IsNullable),
+            InputTypeKind.Float64 => new CSharpType(typeof(double), inputType.IsNullable),
+            InputTypeKind.Float128 => new CSharpType(typeof(decimal), inputType.IsNullable),
+            InputTypeKind.Int32 => new CSharpType(typeof(int), inputType.IsNullable),
+            InputTypeKind.Int64 => new CSharpType(typeof(long), inputType.IsNullable),
+            InputTypeKind.List => new CSharpType(typeof(IList<>), inputType.IsNullable, CreateType(inputType.ValuesType!)),
+            InputTypeKind.ResourceIdentifier => new CSharpType(typeof(ResourceIdentifier), inputType.IsNullable),
+            InputTypeKind.ResourceType => new CSharpType(typeof(ResourceType), inputType.IsNullable),
```

Comment:
I saw that we support AzureLocation, ETag, ResourceType and ResourceIdentifier natively here. How could we have have them from the code model from swagger? Or this is one of the things that only cdal will support?

Reply:
I think those models do not come from swagger or cadl. It is some generated model, current we will not get those directly from incoming model

Reply:
Those are created in process of conversion:

https://github.com/Azure/autorest.csharp/blob/dc7e3fb242ca3d24028eb5db3885f7f16edcfe2f/src/AutoRest.CSharp/Common/Input/CodeModelConverter.cs#L228-L232

Reply:
Good to know, thank you!